reference/events.rst

```diff
@@ -0,0 +1,244 @@
+    {
+        $response = $event->getResponse();
+
+        // ... modify the response object
+    }
+
+The FrameworkBundle registers several listeners:
+
+:class:`Symfony\\Component\\HttpKernel\\EventListener\\ProfilerListener`
+    Collects data for the current request.
+
+:class:`Symfony\\Bundle\\WebProfilerBundle\\EventListener\\WebDebugToolbarListener`
+    Injects the Web Debug Toolbar.
+
+:class:`Symfony\\Component\\HttpKernel\\EventListener\\ResponseListener`
+    Fixes the Response ``Content-Type`` based on the request format.
+
+:class:`Symfony\\Component\\HttpKernel\\EventListener\\EsiListener`
+    Adds a ``Surrogate-Control`` HTTP header when the Response needs to be parsed
+    for ESI tags.
+
+.. seealso::
+
+    Read more on the :ref:`kernel.response event <component-http-kernel-kernel-response>`.
+
+These are the built-in Symfony listeners related to this event:
+
+===================================================================================  ========
+Listener Class Name                                                                  Priority
+===================================================================================  ========
+:class:`Symfony\\Component\\HttpKernel\\EventListener\\EsiListener`                  0
+:class:`Symfony\\Component\\HttpKernel\\EventListener\\ResponseListener`             0
+:class:`Symfony\\Bundle\\SecurityBundle\\EventListener\\ResponseListener`            0
+:class:`Symfony\\Component\\HttpKernel\\EventListener\\ProfilerListener`             -100
+:class:`Symfony\\Bundle\\FrameworkBundle\\EventListener\\TestSessionListener`        -128
+:class:`Symfony\\Bundle\\WebProfilerBundle\\EventListener\\WebDebugToolbarListener`  -128
+:class:`Symfony\\Component\\HttpKernel\\EventListener\\StreamedResponseListener`     -1024
+===================================================================================  ========
+
+``kernel.terminate``
+~~~~~~~~~~~~~~~~~~~~
+
+**Event Class**: :class:`Symfony\\Component\\HttpKernel\\Event\\PostResponseEvent`
+
+The purpose of this event is to perform "heavier" tasks after the response
+was already served to the client.
+
+.. seealso::
+
+    Read more on the :ref:`kernel.terminate event <component-http-kernel-kernel-terminate>`.
+
+This is the built-in Symfony listener related to this event:
+
+=========================================================================  ========
+Listener Class Name                                                        Priority
+=========================================================================  ========
+`EmailSenderListener`_                                                     0
+=========================================================================  ========
+
+
+.. _kernel-kernel.exception:
+
+``kernel.exception``
+~~~~~~~~~~~~~~~~~~~~
+
+**Event Class**: :class:`Symfony\\Component\\HttpKernel\\Event\\GetResponseForExceptionEvent`
+
+The FrameworkBundle registers an
```

Comment:
technically, it is registered by TwigBundle, not by FrameworkBundle (the builtin ExceptionController is provided by TwigBundle as it depends on Twig)